scripts/inference.py

```diff
@@ -142,7 +147,7 @@
 if args.compile:
     print("compiling model")
     # compiling can make first inference pass slow
-    model = torch.compile(model, mode=args.compile_mode)
+    model.forward = torch.compile(model.forward, mode=args.compile_mode)
```

Comment:
was there something in particular that made this necessary? I wonder if we should be using `model.compile()` instead for this kind of thing.

Reply:
yes, when we were using torch.compile(model), we lost the model signature (which we use to check if we can pass position_ids and mask), we could try `model.compile()`, I haven't done that yet for this and will check if this works.

Reply:
Fixed in latest commit to use the underlying model forward